pkg/controllers/workspace_controller.go

```diff
@@ -51,17 +68,26 @@ func (c *WorkspaceReconciler) Reconcile(ctx context.Context, req reconcile.Reque
 	}
 	// TODO apply InferenceSpec
 	// TODO apply TrainingSpec
-	// TODO Update workspace status
-	err = c.setWorkspaceStatusCondition(ctx, workspaceObj, kdmv1alpha1.WorkspaceConditionTypeReady, metav1.ConditionTrue, "Done", "Done")
-	if err != nil {
-		return reconcile.Result{}, err
+	if !apimeta.IsStatusConditionTrue(workspaceObj.Status.Conditions, string(kdmv1alpha1.WorkspaceConditionTypeReady)) {
+		err = c.setWorkspaceStatusCondition(ctx, workspaceObj, kdmv1alpha1.WorkspaceConditionTypeReady, metav1.ConditionTrue,
+			"WorkspaceReady", "workspace is ready")
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 	}
 
 	return reconcile.Result{}, nil
 }
 
 func (c *WorkspaceReconciler) applyWorkspaceResource(ctx context.Context, wObj *kdmv1alpha1.Workspace) error {
 	klog.InfoS("applyWorkspaceResource", "workspace", klog.KObj(wObj))
+	if !apimeta.IsStatusConditionFalse(wObj.Status.Conditions, string(kdmv1alpha1.WorkspaceConditionTypeResourceProvisioned)) {
```

Comment:
You should only update status when there is an OP. remove this.

Reply:
Removed.